File: gdx2d-desktop/src/hevs/gdx2d/demos/simple/DemoAACircles.java
package hevs.gdx2d.demos.simple;

import hevs.gdx2d.components.colors.Palette;
import hevs.gdx2d.lib.GdxGraphics;
import hevs.gdx2d.lib.PortableApplication;

import java.util.Random;

/**
 * Demonstrates the use of anti-aliased circles function
 * 
 * @author Pierre-André Mudry (mui)
 * @version 1.0
 */
public class DemoAACircles extends PortableApplication {

	Random r = new Random();

	// The number of circles to generate
	static final int Q = 5;

	// To store the various circles informations
	int[] x = new int[Q];
	int[] y = new int[Q];
	int[] rad = new int[Q];
	int[] p = new int[Q];

	public DemoAACircles() {
		super(false);
	}

	@Override
	public void onInit() {
		/**
		 * Create all those circles, at various locations and using different
		 * colors from a palette
		 */
		for (int i = 0; i < Q; i++) {
			x[i] = r.nextInt(500);
			y[i] = r.nextInt(500);
			rad[i] = r.nextInt(30) + 10;
			p[i] = r.nextInt(Palette.pastel2.length);
		}
	}

	boolean done = false;

	@Override
	public void onGraphicRender(GdxGraphics g) {

		/**
		 * Draws all those circles. TODO : this is slow and performance should
		 * be improved, using texture?
		 */
		if (!done) {
			g.clear();
			for (int i = 0; i < Q; i++) {
				g.drawAntiAliasedCircle(x[i], y[i], rad[i],
						Palette.pastel2[p[i]]);
			}
			done = false;
			
			g.drawFPS();
			g.drawSchoolLogo();
		}

	}

	public static void main(String[] args) {
		new DemoAACircles();

	}
}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/shaders/simple/DemoAACircles.scala
package ch.hevs.gdx2d.demos.shaders.simple

import java.util.Random

import ch.hevs.gdx2d.components.colors.Palette
import ch.hevs.gdx2d.desktop.DesktopApplication
import ch.hevs.gdx2d.lib.GdxGraphics

/**
 * Demonstrates the use of anti-aliased circles function.
 *
 * @author Pierre-André Mudry (mui)
 * @version 2.0
 */
class DemoAACircles extends DesktopApplication {

  private val rnd = new Random()
  private val Q = 5
  private val x = new Array[Int](Q)
  private val y = new Array[Int](Q)
  private val rad = new Array[Int](Q)
  private val p = new Array[Int](Q)

  override def onInit(): Unit = {
    for (i <- 0 until Q) {
      x(i) = rnd.nextInt(500)
      y(i) = rnd.nextInt(500)
      rad(i) = rnd.nextInt(30) + 10
      p(i) = rnd.nextInt(Palette.pastel2.length)
    }
  }

  override def onGraphicRender(g: GdxGraphics): Unit = {
    g.clear()
    for (i <- 0 until Q) {
      g.drawAntiAliasedCircle(x(i).toFloat, y(i).toFloat, rad(i).toFloat, Palette.pastel2(p(i)))
    }
    g.drawFPS()
    g.drawSchoolLogo()
  }
}

object DemoAACircles {
  def main(args: Array[String]): Unit = {
    new DemoAACircles().launch()
  }
}
